### DISCUSSION TITLE
Storing state in App

### DISCUSSION THREAD

**Discussion by daddycocoaman** [Textualize/textual#4472] [Ideas]
This discussion references https://github.com/Textualize/textual/discussions/4107. It was labeled as answered by the topic creator, but I think it fell short a little bit.

I think it might be useful to have a context dict class for the App (similar to how Typer/Click have one) with a function to request the type of the context item as well. For example:

```
@overloads
get(name, type, default)  - function that also accepts a type, and returns that type. Returns a default if not found.
get(name, type) - function that also accepts a type, and returns that type. Raises an exception if not found.
```

Currently, you can add attributes to the App class you create, but accessing those attributes with type info from a Widget doesn't work because `self.app` returns an `App[object]` and not the created App class.

**Comment by and-viceversa** [comment]
Agreed on this idea and the linked discussion. I settled on a similar simple solution, but also had the impluse to reach for a `click` like Context feature. This seems to work for now but it seems like there should be a better way.

```
class MyApp(App):
    def __init__(self) -> None:
        self.context = {"state1": "", "state2": ""}
        super().__init__()
```
